packages/server/src/config.ts

```diff
@@ -48,27 +59,44 @@ const defaults = {
     "cypress/**/*",
   ],
   includePaths: ["**/*"],
-  devFolder: ".blitz/caches/dev",
-  buildFolder: ".blitz/caches/build",
-  nextBinPatched: "./node_modules/.bin/next-patched",
-  writeManifestFile: true,
 }
 
-export async function normalize(config: ServerConfig): Promise<NormalizedConfig> {
-  const nextBinOrig = await resolveBinAsync("next")
-  const nextBinPatched = await resolveBinAsync("@blitzjs/server", "next-patched")
-  const git = parseChokidarRulesFromGitignore(resolve(process.cwd(), config.rootFolder))
+export async function normalize(
+  config: ServerConfig,
+  dev: boolean = false,
+): Promise<NormalizedConfig> {
+  const rootFolder = resolve(process.cwd(), config.rootFolder)
+  const git = parseChokidarRulesFromGitignore(rootFolder)
 
   return {
     ...config,
-    buildFolder: resolve(config.rootFolder, config.buildFolder ?? defaults.buildFolder),
-    devFolder: resolve(config.rootFolder, config.devFolder ?? defaults.devFolder),
-    ignore: defaults.ignoredPaths.concat(git.ignoredPaths),
-    include: defaults.includePaths.concat(git.includePaths),
-    nextBin: resolve(config.rootFolder, config.interceptNextErrors ? nextBinPatched : nextBinOrig),
+    // -
+    rootFolder,
+    buildFolder: resolve(rootFolder, config.buildFolder ?? defaults.buildFolder),
+    devFolder: resolve(rootFolder, config.devFolder ?? defaults.devFolder),
+    // -
+    isTypescript: config.isTypescript ?? (await getIsTypescript(rootFolder)),
+    watch: config.watch ?? dev,
+    // -
     transformFiles: config.transformFiles ?? transformFiles,
-    watch: config.watch ?? false,
     writeManifestFile: config.writeManifestFile ?? defaults.writeManifestFile,
-    isTypescript: config.isTypescript ?? true,
+    // -
+    ignore: defaults.ignoredPaths.concat(git.ignoredPaths),
+    include: defaults.includePaths.concat(git.includePaths),
+    // -
+    nextBin: await getNextBin(rootFolder, dev),
   }
 }
+
+async function getNextBin(rootFolder: string, dev: boolean = false) {
+  // do not await for both bin-pkg because just one is used at a time
+  const nextBinPkg = dev ? "@blitzjs/server" : "next"
+  const nextBinExec = dev ? "next-patched" : undefined
+  const nextBin = await resolveBinAsync(nextBinPkg, nextBinExec)
+  return resolve(rootFolder, nextBin)
+}
+
+async function getIsTypescript(rootFolder: string) {
+  const fs = await import("fs")
+  return fs.existsSync(join(rootFolder, "tsconfig.json"))
```

Comment:
This doesn't really matter as it is not in stream but why not use a promisified `exists` function like you might from using promisify with fs.exists. If we had to run concurrent processes later that needed this function it wouldn't block the thread that way also why not just import fs up top?

Reply:
- Importing `fs` on top: this is related to `--ts` flag. Once I was thinking about avoiding `fs.existsSync` I was already thinking about `do not import anything unnecessary if you know typescript is set explicitly`. 
- existsSync: it could be definitely done with promisified `exists`

I see now `not importing fs` approach is not very benefical hera as `fs` imported already for other reasons. So I agree to use promisified check and import from fs directly would be better.

I am gonna update this, ok?

Reply:
Done in current commit